### PR TITLE
Enhance Pod API to reflect resources for the executor

### LINKED ISSUE #4707: Add executor resource requirements to PodDefinion
Fixes #4707 

When running pods, marathon will consume more resources than reserved through the containers. These more resources are allocated to the executor. To make this transparent and configurable for the user this must be added to the PodDefinition and the Pod API.

### PR DESCRIPTION
**Pull request by unterstein**
Fixes #4707 

When running pods, marathon will consume more resources than reserved through the containers. These more resources are allocated to the executor. To make this transparent and configurable for the user this must be added to the PodDefinition and the Pod API.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/mesosphere/marathon/pull/4708" target="_blank">ReviewNinja

File: docs/docs/rest-api/public/api/v2/types/resources.raml [hunk 1]
```diff
@@ -36,3 +36,32 @@ types:
...
+
+  ExecutorResources:
+    type: object
+    description: Executor Resource Allocations. Same as Resources but reserved for the pod executor.
+    properties:
+      cpus?:
+        displayName: cpu shares
+        description: The number of CPU shares this pod needs per instance for its executor. This number does not have to be integer, but can be a fraction.
+        type: number
+        minimum: 0.001
```
Comment (aquamatthias) [inline docs/docs/rest-api/public/api/v2/types/resources.raml]: Isn't it possible to use 0 here? Probably more a mesos related question.

File: docs/docs/rest-api/public/api/v2/types/resources.raml [hunk 2]
```diff
@@ -36,3 +36,32 @@ types:
...
+        format: double
+        example: 0.1
+        default: 0.1
+      mem?:
+        displayName: memory shares
+        type: number
+        minimum: 0.001
```
Comment (aquamatthias) [inline docs/docs/rest-api/public/api/v2/types/resources.raml]: No it is not a share. It is MB.
Isn't it possible to use 0 here? Probably more a mesos related question.

File: src/main/scala/mesosphere/marathon/core/pod/PodDefinition.scala [hunk 3]
```diff
@@ -116,4 +117,20 @@ object PodDefinition {
   val DefaultNetworks = Seq.empty[Network]
   val DefaultBackoffStrategy = BackoffStrategy()
   val DefaultUpgradeStrategy = AppDefinition.DefaultUpgradeStrategy
+
+  implicit class ResourcesToExecutorResources(private val resources: Resources) extends AnyVal {
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/pod/PodDefinition.scala]: Please move those classes into a conversion package and implement Reads/Writes.

File: docs/docs/rest-api/public/api/v2/types/resources.raml (hunk 2 above)
Comment (meichstedt) [inline docs/docs/rest-api/public/api/v2/types/resources.raml]: Even if Mesos allows setting 0.001 MB mem, I wouldn't encourage people to do so ;)

File: docs/docs/rest-api/public/api/v2/types/resources.raml [hunk 4]
```diff
@@ -36,3 +36,32 @@ types:
...
+        format: double
+        description: The amount of memory in MB that is needed for the pod instance for its executor
```
Comment (meichstedt) [inline docs/docs/rest-api/public/api/v2/types/resources.raml]: nit: `for the pod instance's executor`

File: docs/docs/rest-api/public/api/v2/types/resources.raml (hunk 2 above)
Comment (unterstein) [inline docs/docs/rest-api/public/api/v2/types/resources.raml]: taken from resources section.

File: src/main/scala/mesosphere/marathon/core/pod/PodDefinition.scala (hunk 3 above)
Comment (unterstein) [inline src/main/scala/mesosphere/marathon/core/pod/PodDefinition.scala]: Reads/Writes from raml->raml?

File: docs/docs/rest-api/public/api/v2/types/resources.raml (hunk 1 above)
Comment (unterstein) [inline docs/docs/rest-api/public/api/v2/types/resources.raml]: Nope, like discussed with @nfnt . Adapted the minimum values to mesos minimum values.

**Comment by aquamatthias** [comment]
retest this please

**Comment by unterstein** [comment]
retest this please

**Comment by unterstein** [comment]
retest this please

**Comment by unterstein** [comment]
retest this please

**Comment by unterstein** [comment]
retest this please

**Comment by unterstein** [comment]
retest this please

**Comment by aquamatthias** [comment]
Thanks!

**Pull request merged**